lib/ext/Infobox/index.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+/**
+ * Parsoid native extension that allows Parsoid to properly render portable infoboxes by delegating rendering
+ * to the MediaWiki PHP parser.
+ */
+
+const {
+	DOMUtils, DOMDataUtils, Promise, parseTokenContentsToDOM
+} = module.parent.require('./extapi.js').versionCheck('^0.10.0');
+
+/**
+ * Inject a placeholder for each <infobox> into the DOM, to be handled at the post-processing stage.
+ *
+ * @param state parsing pipeline context
+ * @param content raw text content of the <infobox> tag
+ * @param attributes attributes set on the <infobox> tag
+ * @return {*|PromiseLike<Document>|Promise<Document>}
+ */
+const toDOM = function (state, content, attributes) {
+	return parseTokenContentsToDOM(state, [], '', '', {
+		wrapperTag: 'aside',
+		pipelineOpts: {
+			extTag: 'infobox',
+		},
+	});
+};
+
+/**
+ * DOM post-processor that operates on the complete HTML5 document constructed by Parsoid from the input content.
+ */
+class PostProcessingPortableInfoboxRenderer {
+	run(node, env, options, atTopLevel) {
+		if (!atTopLevel) {
+			return;
+		}
+
+		const document = node.ownerDocument;
+		const potentialInfoboxNodes = Array.from(document.getElementsByTagName('aside'));
+
+		const actualInfoboxNodes = [];
+
+		potentialInfoboxNodes.forEach(node => {
+			// Only handle actual infoboxes and not any <aside> tags
+			if (DOMUtils.hasTypeOf(node, 'mw:Extension/infobox')) {
+				if (DOMUtils.hasTypeOf(node, 'mw:Transclusion')) {
+					// Found an <infobox> included on the page as a template - we can handle it
+					actualInfoboxNodes.push(node);
+				} else {
+					// Found an <infobox> tag that is not rendered via a template transclusion
+					// It's not really clear what could be done with it - so discard it
+					node.parentNode.removeChild(node);
```

Comment:
aren't we able to render it properly in the editor? when you put <infobox> in article's wikitext it renders just fine